src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -206,6 +205,57 @@ public void on(final PlayerInteractEvent event)
     }
 
     /**
+     * PlayerInteractEvent.EntityInteract handler.
+     * <p>
+     * Check, if a player right clicked an entity.
+     * Deny if:
+     * - If the entity is in colony
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent.EntityInteract event)
+    {
+        final EntityPlayer playerIn = event.getEntityPlayer();
+        if (Configurations.enableColonyProtection && colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            final Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+            if (rank.ordinal() >= Permissions.Rank.FRIEND.ordinal())
```

Comment:
Instead of directly using a rank, check if an action is allowed, I've used ACCESS_HUTS for a lot in the past, but you can always make more. This allows us to easily make this configurable per user/colony in the future.